aiotfm/connection.py

```diff
@@ -95,7 +95,16 @@ def __init__(self, name, client, loop=None):
 				data = await self.socket.recv(length)
 				await self.client.received_data(data, self)
 		except Exception as e:
-			self.client.dispatch('connection_error', self, e)
+			future = self.client.dispatch('connection_error', self, e)
+			if future is not None:
+				# This future is a wrapper for _run_event, which returns True if the event ran successfully, False otherwise.
+				# If it returns False, it already handled the auto_restart.
+				if not await future:
+					return
+			if self.client.auto_restart:
+				await self.client.restart()
```

Comment:
The client should wait few seconds before restarting.

Reply:
It should also trigger an event (on_restart I fit perfectly) to tell the developer the client is restarting.